azure/iac/sql_alert_policy.rego

```diff
@@ -1,91 +1,141 @@
+    not sql_resources.properties.state
+}
+
+azure_attribute_absence["sql_server_security_alert_disabled"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/servers/securityalertpolicies"
+    not resource.properties.state
+}
+
+
 azure_sql_security_alert_disabled["sql_server_security_alert_disabled"] {
     resource := input.resources[_]
     lower(resource.type) == "microsoft.sql/servers"
     sql_resources := resource.resources[_]
     lower(sql_resources.type) == "securityalertpolicies"
-    sql_resources.properties.state == "Disabled"
+    lower(sql_resources.properties.state) == "disabled"
 }
 
 azure_sql_security_alert_disabled["sql_server_security_alert_disabled"] {
     resource := input.resources[_]
     lower(resource.type) == "microsoft.sql/servers/securityalertpolicies"
-    resource.properties.state == "Disabled"
+    lower(resource.properties.state) == "disabled"
 }
 
 sql_server_alert {
     lower(input.resources[_].type) == "microsoft.sql/servers"
+    not azure_attribute_absence["sql_server_security_alert_disabled"]
     not azure_sql_security_alert_disabled["sql_server_security_alert_disabled"]
 }
 
+sql_server_alert = false {
+    azure_attribute_absence["sql_server_security_alert_disabled"]
+}
+
 sql_server_alert = false {
     azure_sql_security_alert_disabled["sql_server_security_alert_disabled"]
 }
 
+sql_server_alert_err = "Security alert for SQL server missing in the resource!" {
+    azure_attribute_absence["sql_server_security_alert_disabled"]
+}
+
 sql_server_alert_err = "Security alert for SQL server is Disabled" {
     azure_sql_security_alert_disabled["sql_server_security_alert_disabled"]
 }
 
 sql_server_alert_metadata := {
-    "Policy Code": "PR-AZR-0103-ARM",
+    "Policy Code": "PR-AZR-0102-ARM",
     "Type": "IaC",
     "Product": "AZR",
     "Language": "ARM template",
-    "Policy Title": "",
-    "Policy Description": "",
+    "Policy Title": "Advanced data security should be enabled on your SQL servers.",
+    "Policy Description": "Advanced data security should be enabled on your SQL servers.",
     "Resource Type": "securityalertpolicies",
     "Policy Help URL": "",
     "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.sql/servers/securityalertpolicies"
 }
 
+# PR-AZR-0103-ARM
 
 default sql_managed_instance_alert = null
 
+azure_attribute_absence["sql_instance_security_alert_disabled"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/managedinstances"
+    sql_resources := resource.resources[_]
+    lower(sql_resources.type) == "securityalertpolicies"
+    not sql_resources.properties.state
+}
+
+azure_attribute_absence["sql_instance_security_alert_disabled"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/managedinstances/securityalertpolicies"
+    not resource.properties.state
+}
+
 azure_sql_security_alert_disabled["sql_instance_security_alert_disabled"] {
```

Comment:
**This rule is unnecessary i believe. please comment it out so that script ignore it. try to find a real world template reference where it shows securityalertpolicies as a child resource type inside microsoft.sql/managedinstances and if found create a separate rule with that.**